src/Plug-ins/RemoteAppExecution/DicomDeidentifier.cs

```diff
@@ -53,52 +53,59 @@ public class DicomDeidentifier : IOutputDataPlugIn
+                newRecord.OriginalValues.Add(DicomTag.StudyInstanceUID.ToString(), studyInstanceUid);
+                newRecord.OriginalValues.Add(DicomTag.SeriesInstanceUID.ToString(), seriesInstanceUid);
+                newRecord.OriginalValues.Add(DicomTag.SOPInstanceUID.ToString(), dicomFile.Dataset.GetSingleValue<string>(DicomTag.SOPInstanceUID));
 
-            dicomFile.Dataset.AddOrUpdate(DicomTag.StudyInstanceUID, newRecord.StudyInstanceUid);
-            dicomFile.Dataset.AddOrUpdate(DicomTag.SeriesInstanceUID, newRecord.SeriesInstanceUid);
-            dicomFile.Dataset.AddOrUpdate(DicomTag.SOPInstanceUID, newRecord.SopInstanceUid);
+                dicomFile.Dataset.AddOrUpdate(DicomTag.StudyInstanceUID, newRecord.StudyInstanceUid);
+                dicomFile.Dataset.AddOrUpdate(DicomTag.SeriesInstanceUID, newRecord.SeriesInstanceUid);
+                dicomFile.Dataset.AddOrUpdate(DicomTag.SOPInstanceUID, newRecord.SopInstanceUid);
 
-            foreach (var tag in tags)
-            {
-                if (tag.Equals(DicomTag.StudyInstanceUID) ||
-                    tag.Equals(DicomTag.SeriesInstanceUID) ||
-                    tag.Equals(DicomTag.SOPInstanceUID))
+                foreach (var tag in tags)
                 {
-                    continue;
-                }
+                    if (tag.Equals(DicomTag.StudyInstanceUID) ||
+                        tag.Equals(DicomTag.SeriesInstanceUID) ||
+                        tag.Equals(DicomTag.SOPInstanceUID))
+                    {
+                        continue;
+                    }
 
-                if (dicomFile.Dataset.TryGetString(tag, out var value))
-                {
-                    newRecord.OriginalValues.Add(tag.ToString(), value);
-                    var newValue = Utilities.GetTagProxyValue<string>(tag);
-                    if (newValue != null)
+                    if (dicomFile.Dataset.TryGetString(tag, out var value))
                     {
-                        dicomFile.Dataset.AddOrUpdate(tag, newValue);
-                        _logger.ValueChanged(tag.ToString(), value, newValue);
+                        newRecord.OriginalValues.Add(tag.ToString(), value);
+                        var newValue = Utilities.GetTagProxyValue<string>(tag);
+                        if (newValue != null)
+                        {
+                            dicomFile.Dataset.AddOrUpdate(tag, newValue);
+                            _logger.ValueChanged(tag.ToString(), value, newValue);
+                        }
                     }
                 }
-            }
 
-            await repository.AddAsync(newRecord).ConfigureAwait(false);
+                await repository.AddAsync(newRecord).ConfigureAwait(false);
 
-            return (dicomFile, exportRequestDataMessage);
+                return (dicomFile, exportRequestDataMessage);
+            }
+            catch (Exception)
```

Comment:
Do we need to try catch this block if were throwing anyway?

Reply:
see earlier comment to Sam :)